RecoEgamma/EgammaTools/plugins/EGExtraInfoModifierFromDB.cc

```diff
@@ -0,0 +1,730 @@
+  pho_vmaps.clear();
+  
+  if( !e_conf.tok_electron_src.isUninitialized() ) {
+    edm::Handle<edm::View<pat::Electron> > eles;
+    evt.getByToken(e_conf.tok_electron_src, eles);
+    
+    for( unsigned i = 0; i < eles->size(); ++i ) {
+      edm::Ptr<pat::Electron> ptr = eles->ptrAt(i);
+      eles_by_oop[ptr->originalObjectRef().key()] = ptr;
+    }    
+  }
+
+  get_product(evt, e_conf.tok_sigmaIetaIphi, ele_vmaps);
+  get_product(evt, e_conf.tok_eMax,          ele_vmaps);
+  get_product(evt, e_conf.tok_e2nd,          ele_vmaps);
+  get_product(evt, e_conf.tok_eTop,          ele_vmaps);
+  get_product(evt, e_conf.tok_eBottom,       ele_vmaps);
+  get_product(evt, e_conf.tok_eLeft,         ele_vmaps);
+  get_product(evt, e_conf.tok_eRight,        ele_vmaps);
+  get_product(evt, e_conf.tok_clusterMaxDR	       , ele_vmaps);
+  get_product(evt, e_conf.tok_clusterMaxDRDPhi     , ele_vmaps);
+  get_product(evt, e_conf.tok_clusterMaxDRDEta     , ele_vmaps);
+  get_product(evt, e_conf.tok_clusterMaxDRRawEnergy, ele_vmaps);
+  get_product(evt, e_conf.tok_clusterRawEnergy0    , ele_vmaps);
+  get_product(evt, e_conf.tok_clusterRawEnergy1    , ele_vmaps);
+  get_product(evt, e_conf.tok_clusterRawEnergy2    , ele_vmaps);
+  get_product(evt, e_conf.tok_clusterDPhiToSeed0   , ele_vmaps);
+  get_product(evt, e_conf.tok_clusterDPhiToSeed1   , ele_vmaps);
+  get_product(evt, e_conf.tok_clusterDPhiToSeed2   , ele_vmaps);
+  get_product(evt, e_conf.tok_clusterDEtaToSeed0   , ele_vmaps);
+  get_product(evt, e_conf.tok_clusterDEtaToSeed1   , ele_vmaps);
+  get_product(evt, e_conf.tok_clusterDEtaToSeed2   , ele_vmaps);
+  get_product(evt, e_conf.tok_iPhi,          ele_int_vmaps);
+  get_product(evt, e_conf.tok_iEta,          ele_int_vmaps);
+  get_product(evt, e_conf.tok_cryPhi,        ele_vmaps);
+  get_product(evt, e_conf.tok_cryEta,        ele_vmaps);
+  
+  if( !ph_conf.tok_photon_src.isUninitialized() ) {
+    edm::Handle<edm::View<pat::Photon> > phos;
+    evt.getByToken(ph_conf.tok_photon_src,phos);
+  
+    for( unsigned i = 0; i < phos->size(); ++i ) {
+      edm::Ptr<pat::Photon> ptr = phos->ptrAt(i);
+      phos_by_oop[ptr->originalObjectRef().key()] = ptr;
+    }
+  }
+   
+  get_product(evt, ph_conf.tok_sigmaIetaIphi, pho_vmaps);
+  get_product(evt, ph_conf.tok_sigmaIphiIphi, pho_vmaps);
+  get_product(evt, ph_conf.tok_e2x5Max,       pho_vmaps);
+  get_product(evt, ph_conf.tok_e2x5Left,      pho_vmaps);
+  get_product(evt, ph_conf.tok_e2x5Right,     pho_vmaps);
+  get_product(evt, ph_conf.tok_e2x5Top,       pho_vmaps);
+  get_product(evt, ph_conf.tok_e2x5Bottom,    pho_vmaps);
+  
+  if (autoDetectBunchSpacing_) {
+    if (evt.isRealData()) {
+      edm::RunNumber_t run = evt.run();
+      if (run == 178003 ||
+	  run == 178004 ||
+	  run == 209089 ||
+	  run == 209106 ||
+	  run == 209109 ||
+	  run == 209146 ||
+	  run == 209148 ||
+	  run == 209151 || 
+	  run > 253000) {
+	bunchspacing_ = 25;
+      } else {
+	bunchspacing_ = 50;
```

Comment:
make this in sync with 
https://github.com/cms-sw/cmssw/blob/CMSSW_7_4_X/RecoParticleFlow/PFClusterProducer/src/PFClusterEMEnergyCorrector.cc#L140
[this temporary hack that we introduced is becoming a norm :( ]